core/src/main/java/com/linecorp/armeria/internal/server/CorsHeaderUtil.java

```diff
@@ -0,0 +1,146 @@
+package com.linecorp.armeria.internal.server;
+
+import com.google.common.base.Strings;
+import com.linecorp.armeria.common.*;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.cors.CorsConfig;
+import com.linecorp.armeria.server.cors.CorsPolicy;
+import com.linecorp.armeria.server.cors.CorsService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class CorsHeaderUtil {
+
+    private static final Logger logger = LoggerFactory.getLogger(CorsHeaderUtil.class);
+
+    public static final String ANY_ORIGIN = "*";
+    public static final String NULL_ORIGIN = "null";
+
+    /**
+     * Emit CORS headers if origin was found.
+     *
+     * @param req     the HTTP request with the CORS info
+     * @param headers the headers to modify
+     */
+    public static void setCorsResponseHeaders(ServiceRequestContext ctx, HttpRequest req,
+                                              ResponseHeadersBuilder headers, CorsService corsService) {
```

Comment:
Can we accept a `CorsConfig` as a parameter instead of `CorsService`? This method doesn't really depend on `CorsService` but only on `CorsConfig`.

Reply:
Sure. I fixed it.